scripts/us_census/enhanced_tmcf/process_etmcf.py

```diff
@@ -0,0 +1,286 @@
+                    print(
+                        f"... geoId: {row_dict[GEO_ID_COLUMN]} does not exist in DC. Skipping input row # {row_counter} in file {original_csv_path}"
+                    )
+                    continue
+                else:
+                    row_dict[GEO_ID_COLUMN] = geo_ids_to_dcids[
+                        row_dict[GEO_ID_COLUMN]]
+
+            # To the row_dict, add the key/values corresponding to the new columns.
+            for col in new_cols:
+                vals = [row_dict[k] for k in col.ordered_csv_column_names]
+                val_str = col.format.format(*vals)
+                row_dict.update({col.name: val_str})
+
+            output_rows.append(row_dict)
+
+    with open(output_csv_path, 'w', newline='') as output_csv:
+        writer = csv.DictWriter(output_csv, fieldnames=csv_column_names)
+        writer.writeheader()
+        for row_dict in output_rows:
+            writer.writerow(row_dict)
+
+
+def process_enhanced_tmcf(input_folder, output_folder, etmcf_filename,
+                          input_csv_filename, output_tmcf_filename,
+                          output_csv_filename) -> None:
+    """Processed the enhanced TMCF file and input CSV. Produces the traditional TMCF and modified CSV."""
+    if not os.path.exists(input_folder):
+        raise Exception(
+            f"input_folder should exist.\nInput Folder: {input_folder}")
+    if not os.path.exists(output_folder):
+        raise Exception(
+            f"output_folder should exist.\nOutput Folder: {output_folder}")
+
+    etmcf_filepath = os.path.join(input_folder, etmcf_filename + ".tmcf")
+    input_csv_filepath = os.path.join(input_folder, input_csv_filename + ".csv")
+
+    if not os.path.exists(etmcf_filepath):
+        raise Exception(
+            f"Enhanced TMCF file should exist.\nE_TMCF file: {etmcf_filepath}")
+    if not os.path.exists(input_csv_filepath):
+        raise Exception(
+            f"Input CSV file should exist.\nInput CSV file: {input_csv_filepath}"
+        )
+
+    # If the input_csv_filename has the 'geoId' column, pre-process to get all geoIds which
+    # do not yet exist in DC.
+    geo_ids = _get_geoId_column(input_csv_filepath)
+    geo_ids_to_dcids = _census_geoId_to_dcid(geo_ids)
+    geo_ids_not_found = _get_places_not_found(geo_ids)
+
+    # new_csv_columns is a list of new columns to add to produce the processed CSV.
+    new_csv_columns: List[NewCSVColumn] = []
+
+    T = ""
+    data_files = ""
+    tmcf_out = os.path.join(output_folder, output_tmcf_filename + ".tmcf")
+    csv_out = os.path.join(output_folder, output_csv_filename + ".csv")
+
+    # Process the enhanced TMCF to identify the new columns to be introduced in the csv.
+    with open(etmcf_filepath, 'r') as etmcf, open(tmcf_out, 'w') as out_tmcf:
+        lines = etmcf.readlines()
+
+        output_started = False
+        for line in lines:
+            out_line = line
+            if line.startswith("Namespace:"):
```

Comment:
not sure if we need to be stricter with ensuring `Namespace` is immediately before `T=` line, but this is good for now. add a TODO?

Reply:
Added a TODO